config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/Entity.java

```diff
@@ -14,7 +14,7 @@
 import java.util.HashMap;
 import java.util.Map;
 
-public class Entity {
+public class Entity<T> {
```

Comment:
Can we avoid generic? Instead, just have a variable of type `GatewayEntity`. Whoever wants the object, it can cast it using `instanceof` check. Generic doesn't help much with respect to the way we are accessing this object.

Reply:
Done